### PR TITLE
feat(revwalks): adding gitutils implementation

### LINKED ISSUE #4: Start using `gitutils` package
Let's start comparing with https://github.com/mergestat/gitutils

### PR DESCRIPTION
**Pull request by ramiroJCB**
Chaning previous git log walk implementation for our newest  gitutils one.

Closes #4

### CODE REVIEW

File: revwalks_stats.go [hunk 1]
```diff
@@ -56,19 +58,32 @@ func GoGitRevWalkStats(repoPath string) error {
 }
 
 func GitCLIRevWalkStats(repoPath string) error {
-	res, err := gitlog.Exec(context.Background(), repoPath, "", nil)
+	iter, err := gitlog.Exec(context.Background(), repoPath, gitlog.WithStats(true))
+
 	if err != nil {
+		fmt.Println("@hejrer")
```
Comment (patrickdevivo) [inline revwalks_stats.go]:
```suggestion
```

**Review by patrickdevivo — CHANGES_REQUESTED** [review]

File: revwalks_stats.go (hunk 1 above)
Comment (ramiroJCB) [reply revwalks_stats.go]: Lol

File: revwalks.go [hunk 2]
```diff
@@ -44,15 +47,31 @@ func GoGitRevWalk(repoPath string) error {
...
+			if errors.Is(err, io.EOF) {
+				break
+			}
+
+			log.Fatal(err)
+		} else {
+
+			if commit == nil {
+				break
+			}
```
Comment (patrickdevivo) [inline revwalks.go]:
```suggestion
```

File: revwalks_stats.go [hunk 3]
```diff
@@ -56,19 +59,35 @@ func GoGitRevWalkStats(repoPath string) error {
...
+		if commit, err := iter.Next(); err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+
+			log.Fatal(err)
+		} else {
+			if commit == nil {
+				break
+			}
```
Comment (patrickdevivo) [inline revwalks_stats.go]:
```suggestion
```

**Review by patrickdevivo — CHANGES_REQUESTED** [review]

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**